master/security/icmp-ping.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Use ICMP/ping in policy rules
+---
+
+### Big picture
+
+Use Calico network policy to allow and deny ICMP/ping messages. 
+
+### Value
+
+The **Internet Control Message Protocol (ICMP)** provides valuable network diagnostic functions, but it can also be used maliciously. Attackers can use it to learn about your network, or for DoS attacks. Using Calico network policy, you can control where ICMP is used. For example, you can:
+
+- Allow ICMP ping, but only within your cluster 
+- Allow ICMP for pods launched by operators for diagnostic purposes, but block other uses
+- Temporarily enable ICMP to diagnose a problem, then disable it after the problem is resolved
+- Deny/allow ICMPv4 and/or ICMPv6
+
+### Features
+
+This how-to guide uses the following Calico features:
+
+**GlobalNetworkPolicy** or **NetworkPolicy** with:
+- Positive/negative match criteria for ICMP
+- Protocol match for ICMPv4 and ICMPv6
+- Match on ICMP type and code
+
+### Concepts
+
+#### ICMP packet type and code
+
+Calico network policy also lets you deny and allow specific parts of the ICMP packet for fine-grain control. For example, you can specify ICMP type 5, code 2. For details, see [ICMP type and code](https://en.wikipedia.org/wiki/Internet_Control_Message_Protocol#Control_messages).
+
+### How to
+
+- [Deny all ICMP](#deny-all-icmp)
+- [Allow ICMP ping only within a cluster](#allow-icmp-ping-only-within-a-cluster)
+- [Allow ICMP matching protocol type and code](#allow-icmp-matching-protocol-type-and-code)
+
+#### Deny all ICMP
+
+In this example, we introduce a "deny all ICMP" GlobalNetworkPolicy. Keep in mind that this policy **blocks all traffic to and from any workload in the cluster**. If your ultimate goal is to allow some traffic, have your regular "allow" policies in place before applying the global deny-all policy.
```

Comment:
> Keep in mind that this policy **blocks all traffic to and from any workload in the cluster**

This wording isn't quite right, since it isn't actually denying all traffic. We should say "Keep in mind that this policy selects all workloads in the cluster, and thus enables a default deny for all workloads in addition to the explicit ICMP deny rules specified in the policy" - perhaps linking to the default deny documentation as well.

Reply:
Fixed.